### PR TITLE
Refactoring AccordionPanel to use hooks

### PR DESCRIPTION
**Pull request by VictorKunzler**
**What does this PR do?**
Refactor /src/js/components/AccordionPanel/AccordionPanel.js using Hooks

**Where should the reviewer start?**
/src/js/components/AccordionPanel/AccordionPanel.js

**What testing has been done on this PR?**
Running yarn test and yarn storybook and tested the component behavior.

**How should this be manually tested?**
Running yarn test

**Any background context you want to provide?**
N/A

**What are the relevant issues?**
#3394

**Screenshots (if appropriate)**
N/A

**Do the grommet docs need to be updated?**
No

**Should this PR be mentioned in the release notes?**
No

**Is this change backwards compatible or is it a breaking change?**
Backwards compatible

### CODE REVIEW

File: src/js/components/AccordionPanel/AccordionPanel.js [hunk 1]
```diff
@@ -14,133 +14,125 @@ import { withForwardRef } from '../hocs';
...
-                )}
-              </Button>
-              <Box border={theme.accordion.border}>
-                {animate ? (
-                  <Collapsible open={active}>{children}</Collapsible>
-                ) : (
-                  active && children
-                )}
-              </Box>
+const onHandleMouseOver = ({ onMouseOver, theme, setHover }) => (...args) => {
```
Comment (oorestisime) [inline src/js/components/AccordionPanel/AccordionPanel.js]: any particular reason to have those functions outside of the component? the code will be a bit simpler without needing to provide those parameters in all the functions. not sure the performance gains are even notable to justify having them outside. thoughts?

Comment (VictorKunzler) [reply src/js/components/AccordionPanel/AccordionPanel.js]: In my opinion the better solution always is the component just render the element, all the others logics should be outside the component.

Comment (ShimiSun) [reply src/js/components/AccordionPanel/AccordionPanel.js]: I agree with @oorestisime, l think the code could be more readable and clean having the functions inside the component.

File: src/js/components/AccordionPanel/AccordionPanel.js [hunk 2]
```diff
@@ -60,6 +32,34 @@ const AccordionPanel = ({
     theme,
   );
 
+  const onHandleMouseOver = () => (...args) => {
```
Comment (oorestisime) [inline src/js/components/AccordionPanel/AccordionPanel.js]: may i suggest to remove the () and in the caller to just add `onHandleMouseOver` seems a tiny bit more readable. same case with the rest of them. thanks for taking care of the changes.

Comment (VictorKunzler) [reply src/js/components/AccordionPanel/AccordionPanel.js]: Fixed

**Comment by ShimiSun** [comment]
Thank you for your contributions! 🎉

**Pull request merged**